### DISCUSSION TITLE
Yoda installation without ansible

### DISCUSSION THREAD

**Discussion by ll4strw** [UtrechtUniversity/yoda#240] [Q&A] [answered]
Hello, is there an easy way to install yoda on GNU/Linux without ansible? Do you guys have a human readable playbook that I could follow? Thanks in advance.

**Comment by stsnel** [comment]
Hello Leonardo,

At present, the only supported way to install Yoda is using the Ansible playbook. Although it is technically possible to read the Ansible playbook and perform each step manually, that would be time-consuming and prone to errors. Without any automation, it would also be hard to maintain an environment efficiently. So we would recommend against deploying Yoda that way.

Can you tell a bit more about the reason you don't want to use Ansible?

**Reply by ll4strw** [reply]
Hi Sietse, 
thanks for your answer.  Our set up here does not use ansible for automated deployment. Certainly I could easily set up  a test yoda environment based on ansible, but then I will have problems later on if we decided to move it in production. 
Are there, by any chance, plans to support installations via containers? irods can be deployed using containers after all. Cheers, L.

**Comment by stsnel** [comment, marked as answer]
We are currently working on a containerized setup, but this is intended mainly for local development and testing. If you like, we can let you know when the first development version is available.
 
At the moment, we do not have plans to change Yoda to run iRODS in containers on production environments. Based on our experiences with troubleshooting iRODS, we think it is good to keep the setup as simple as possible. Not using containers gives us more flexibility with troubleshooting and reduces the number of moving parts that can make it harder to diagnose problems.

Depending on your goals and requirements, using a Yoda environment hosted by Surf (https://servicedesk.surf.nl/wiki/display/WIKI/Yoda+Hosting) might be an alternative to consider.

**Comment by ll4strw** [comment]
Yes, please let me know when the first development version of your containerized setup is ready. 
I was aware of the Yoda environments hosted at SURF, but we have our own iRODS instance which is totally uncoupled from SURF. Thanks anyway.

**Reply by stsnel** [reply]
We have just completed an initial version of a containerized setup for development and testing. Instructions about how to use it are available at https://github.com/UtrechtUniversity/yoda/blob/development/docs/development/docker-setup.md . This page also contains additional information about current status of the setup, intended use and limitations.

**Reply by dalito** [reply]
When I try this docker setup, it fails because the images are not accessible. Is this intended? So should I build them locally?

```
λ docker-compose pull
[+] Pulling 22/22
(...)
3.1s WARNING: Some service image(s) must be built from source by running:
    docker compose build provider eus public mailpit datacite_mock portal davrods
7 errors occurred:
        * Error response from daemon: Head "https://ghcr.io/v2/utrechtuniversity/yoda-portal/manifests/dev-1.9": denied: denied
        * Error response from daemon: Head "https://ghcr.io/v2/utrechtuniversity/yoda-mailpit/manifests/dev-1.9": denied: denied
        * Error response from daemon: Head "https://ghcr.io/v2/utrechtuniversity/yoda-web-mock/manifests/dev-1.9": denied: denied
        * Error response from daemon: Head "https://ghcr.io/v2/utrechtuniversity/yoda-provider/manifests/dev-1.9": denied: denied
        * Error response from daemon: Head "https://ghcr.io/v2/utrechtuniversity/yoda-eus/manifests/dev-1.9": denied: denied
        * Error response from daemon: Head "https://ghcr.io/v2/utrechtuniversity/yoda-public/manifests/dev-1.9": denied: denied
        * Error response from daemon: Head "https://ghcr.io/v2/utrechtuniversity/davrods/manifests/dev-1.9": denied: denied
```

**Reply by stsnel** [reply]
You should indeed be able to download the images from GHCR. Are you currently logged in on GHCR? If so, can you try to log out before you pull (`docker logout ghcr.io`)? If it works after that, the cause might be that your token doesn't have a  read:packages scope.

**Reply by dalito** [reply]
Thanks for the quick reply. My problem was that I forgot to create and set a GH token for package registry access. I was not aware that this [is required](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-docker-registry). With the token, I could set up the system as described in the docs. Thanks again!
